vnext/Shared/Networking/OriginPolicyHttpFilter.cpp

```diff
@@ -0,0 +1,638 @@
+  bool withCredentials = winrt::get_self<RequestArgs, IInspectable>(iRequestArgs)->WithCredentials;
+  bool requestMethodAllowed = false;
+  for (const auto &method : controlValues.AllowedMethods) {
+    if (L"*" == method) {
+      if (!withCredentials) {
+        requestMethodAllowed = true;
+        break;
+      }
+    } else if (boost::iequals(method, request.Method().ToString())) {
+      requestMethodAllowed = true;
+      break;
+    }
+  }
+  if (!requestMethodAllowed)
+    throw hresult_error{
+        E_INVALIDARG,
+        L"Method [" + request.Method().ToString() +
+            L"] is not allowed by Access-Control-Allow-Methods in preflight response.\\n"};
+
+  // Check if request headers are allowed
+  // if (!IsCrossOriginRequestHeadersAllowed(requestHeaders, allowedHeadersList, withCredentials, /*out*/ errorText))
+  // See https://fetch.spec.whatwg.org/#cors-preflight-fetch, section 4.8.7.6-7
+  // Check if the header should be allowed through wildcard, if the request does not have credentials.
+  bool requestHeadersAllowed = false;
+  if (!withCredentials && controlValues.AllowedHeaders.find(L"*") != controlValues.AllowedHeaders.cend()) {
+    // "Authorization" header cannot be allowed through wildcard alone.
+    // "Authorization" is the only member of https://fetch.spec.whatwg.org/#cors-non-wildcard-request-header-name.
+    if (request.Headers().HasKey(L"Authorization") && controlValues.AllowedHeaders.find(L"Authorization") == controlValues.AllowedHeaders.cend())
+      throw hresult_error{
+          E_INVALIDARG,
+          L"Request header field [Authorization] is not allowed by Access-Control-Allow-Headers in preflight response.\\n"};
+
+    requestHeadersAllowed = true;
+  }
+  if (!requestHeadersAllowed) {
+    // Forbidden headers are excluded from the JavaScript layer.
+    // User agents may use these headers internally.
+    // TODO: CorsUnsafeNotForbiddenRequestHeaderNames(requestHeaders);
+    const set unsafeNotForbidenHeaderNames = CorsUnsafeNotForbiddenRequestHeaderNames(request.Headers());
+    for (const auto name : unsafeNotForbidenHeaderNames) {
+      if (controlValues.AllowedHeaders.find(name) == controlValues.AllowedHeaders.cend())
+        throw hresult_error{
+            E_INVALIDARG,
+            L"Request header field [" + to_hstring(name) +
+                L"]is not allowed by Access-Control-Allow-Headers in preflight response.\\n"};
+    }
+  }
+
+  // #9770 - insert into preflight cache
+}
+
+// See 10.7.4 of https://fetch.spec.whatwg.org/#http-network-or-cache-fetch
+void OriginPolicyHttpFilter::ValidateResponse(HttpResponseMessage const &response) const {
+  /*
+    _Inout_ std::map<std::wstring, std::wstring>& responseHeaders,
+    NetworkingSecurityPolicy validatedSecurityPolicy,
+    const std::vector<std::wstring>& exposeHeaders,
+    bool withCredentials,
+    _Out_ std::string& errorText) noexcept
+  */
```

Comment:
remove?

Reply:
Yep.

Sorry, the PR is not fully "clean" yet.